graph_adventure/adv.py

```diff
@@ -15,6 +15,27 @@
 # FILL THIS IN
 traversalPath = []
 
+class TraversalGraph:
+  def __init__(self):
+    self.current_room = 0
+    self.visited_rooms = {
+      0: {'n': '?', 's': '?', 'w': '?', 'e': '?'}
+    }
+
+  def pick_unexplored(self):
+    exits = player.currentRoom.getExits()
+    unexplored = []
+    for e in exits:
+      if self.visited_rooms[player.currentRoom.id][e] == '?':
+        unexplored.append(e)
+    if len(unexplored) == 0:
+      return None
```

Comment:
love how you implemented randomness into your algorithm, the way you went about this problem shows how knowledgable you are in the CS fundamentals